keps/sig-storage/0661-statefulset-volume-resize/README.md

```diff
@@ -0,0 +1,808 @@
+as listed in [KEP 1790](../1790-recover-resize-failure/README.md).
+
+## Motivation
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Stable persistent volumes can be created by specifying `volumeClaimTemplates` in `StatefulSet`
+[specification](https://kubernetes.io/docs/concepts/workloads/controllers/StatefulSet/).
+Currently, StatefulSets do not allow the associated volumes to be resized directly by modifying the
+`volumeClaimTemplates`. Instead users have to modify each PVC one by one to achieve this. In such cases, 
+when the statefulset scales up, the new PVC(s) will be created with the older size and this again needs 
+manual intervention. The intent of this proposal is to avoid this user overhead and simplify the 
+process of resizing StatefulSet volumes.
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+Allow users to resize the volumes associated with `StatefulSet` by modifying the size of
+PVC in the `volumeClaimTemplate`.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+This proposal does not try to address the following scenarios but relies on the
+underlying PVC resize logic to take the appropriate action or indicate error as needed.
+- Shrinking of volume is not allowed.
+- Few environments require Pod restarts in order
+to get access to the resized volume.
+- Modifying the storage class of the volumeClaimTemplate
+- Resizing a PVC associated with a statefulset directly by modifying the PVC spec will not affect the 
+  Statefulset `volumeClaimTemplates`. 
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+Before we proceed to the proposal it is imperative to layout the various components involved
+in creating and managing the persistent volumes associated with `StatefulSet`. 
+
+- StatefulSet controller creates the PVCs from `volumeClaimTemplates` and associates them with the pods
+spawned from it. Reconciliation logic runs for every change made to the StatefulSet
+`Spec` and `Status` reflects the progress towards making the real world state equal to the
+`Spec`.
+- The admission controller in the API server is responsible for restricting the fields
```

Comment:
nit: it's not an admission controller - it's just the validation logic

Reply:
Changed it